src/fidesops/graph/config.py

```diff
@@ -298,46 +366,49 @@ class Collection(BaseModel):
     fields: List[Field]
     # an optional list of collections that this collection must run after
     after: Set[CollectionAddress] = set()
-    field_dict: Dict[str, Field] = {}
+
+    field_dict: Dict[FieldPath, Field] = {}
 
     def __init__(self, **kwargs: Dict[str, Any]) -> None:
         super().__init__(**kwargs)
-        self.field_dict = {f.name: f for f in self.fields}
+        self.field_dict = self._collect_matching(lambda f: True)
+
+    def _collect_matching(self, f: Callable[[Field], bool]) -> Dict[FieldPath, Field]:
+        matches = map(lambda field: field.collect_matching(f), self.fields)
+        return merge_dicts(*matches)
 
     def references(
         self,
-    ) -> Dict[str, List[Tuple[FieldAddress, Optional[EdgeDirection]]]]:
+    ) -> Dict[FieldPath, List[Tuple[FieldAddress, Optional[EdgeDirection]]]]:
         """return references from fields in this collection to fields in any other"""
-        flds_w_ref = filter(lambda f: f.references, self.fields)
-        return {f.name: f.references for f in flds_w_ref}
+        return {k: v.references for k, v in self.field_dict.items() if v.references}
 
-    def identities(self) -> Dict[str, Tuple[str, ...]]:
+    def identities(self) -> Dict[FieldPath, Tuple[str, ...]]:
         """return identity pointers included in the table"""
-        flds_w_ident = filter(lambda f: f.identity, self.fields)
-        return {f.name: f.identity for f in flds_w_ident}
+        return {k: v.identity for k, v in self.field_dict.items() if v.identity}
 
-    def field(self, name: str) -> Optional[Field]:
+    def field(self, key: FieldPath) -> Optional[Field]:
```

Comment:
Should we have another name for this "key" variable we're introducing everywhere that is a FieldPath object, maybe "field_path" instead? I'm afraid this might get confused with the other key variable we use everywhere that is a "FidesOpsKey" type.

Reply:
y, again, remnant of previous name.  FieldPath I thought is more descriptive, a few "key" refs floating around still...